Objects/structseq.c

```diff
@@ -409,16 +395,51 @@ PyStructSequence_InitType(PyTypeObject *type, PyStructSequence_Desc *desc)
 PyTypeObject*
 PyStructSequence_NewType(PyStructSequence_Desc *desc)
 {
-    PyTypeObject *result;
+    PyObject* type;
+    PyObject* bases;
+    PyMemberDef* members;
+
+    PyType_Spec* spec = PyMem_NEW(PyType_Spec, 1);
+    spec->name = desc->name;
+    spec->basicsize = sizeof(PyStructSequence) - sizeof(PyObject *);
+    spec->itemsize = sizeof(PyObject *);
+    spec->flags = Py_TPFLAGS_DEFAULT;
+    spec->slots = PyMem_NEW(PyType_Slot, 7);
+
+    spec->slots[0].slot = Py_tp_dealloc;
+    spec->slots[0].pfunc = (destructor)structseq_dealloc;
+
+    spec->slots[1].slot = Py_tp_repr;
+    spec->slots[1].pfunc = (reprfunc)structseq_repr;
 
-    result = (PyTypeObject*)PyType_GenericAlloc(&PyType_Type, 0);
-    if (result == NULL)
+    spec->slots[2].slot = Py_tp_doc;
+    spec->slots[2].pfunc = (void*)desc->doc;
+
+    spec->slots[3].slot = Py_tp_methods;
+    spec->slots[3].pfunc = structseq_methods;
+
+    spec->slots[4].slot = Py_tp_new;
+    spec->slots[4].pfunc = structseq_new;
+
+    members = initialize_members(desc);
+    if (members == NULL)
         return NULL;
-    if (PyStructSequence_InitType2(result, desc) < 0) {
-        Py_DECREF(result);
+    spec->slots[5].slot = Py_tp_members;
+    spec->slots[5].pfunc = members;
+
+    spec->slots[6].slot = 0;
+    spec->slots[6].pfunc = 0;
+
+    bases = PyTuple_Pack(1, &PyTuple_Type);
```

Comment:
I think 'bases' is a new tuple.  Does PyType_FromSpecWithBases() "steal" a reference to it?  I would guess not so you should have a Py_DECREF(bases) somewhere.

Reply:
Nope, there's no reference stealing. I totally missed the decref here. This has been fixed in the latest commit!